core/common/boot/src/main/java/org/eclipse/edc/boot/system/DependencyGraph.java

```diff
@@ -127,24 +144,95 @@ public List<InjectionContainer<ServiceExtension>> of(List<ServiceExtension> exte
                     .forEach(injectionPoint -> container.getInjectionPoints().add(injectionPoint));
         });
 
-        if (!unsatisfiedInjectionPoints.isEmpty()) {
-            var message = "The following injected fields or values were not provided or could not be resolved:\n";
-            message += unsatisfiedInjectionPoints.entrySet().stream()
-                    .map(entry -> String.format("%s is missing \n  --> %s", entry.getKey(), String.join("\n  --> ", entry.getValue().stream().map(Object::toString).toList()))).collect(Collectors.joining("\n"));
-            throw new EdcInjectionException(message);
-        }
+        sort.sort(injectionContainers);
 
-        if (!unsatisfiedRequirements.isEmpty()) {
-            var message = String.format("The following @Require'd features were not provided: [%s]", String.join(", ", unsatisfiedRequirements));
-            throw new EdcException(message);
-        }
+        return new DependencyGraph(injectionContainers, unsatisfiedInjectionPoints, unsatisfiedRequirements);
+    }
 
-        sort.sort(injectionContainers);
+    /**
+     * Returns all {@link InjectionPoint}s for a particular extension class. These include all types of injection points.
+     *
+     * @param serviceClass The extension class
+     * @return A (potentially empty) list of injection points.
+     */
+    public List<InjectionPoint<ServiceExtension>> getDependenciesOf(Class<? extends ServiceExtension> serviceClass) {
```

Comment:
yes, the idea was to make the DG into more like a graph, with methods to match. It would make the DG more usable, esp. in runtimes other than the `BaseRuntime`

Reply:
I had a version where the `DependencyGraph` is an interface, and _one_ implementation is using EDC-internal dependency injection. Other implementations could use other DI frameworks like Guice, which would make the DI independent from the runtime. I ultimately abandoned that, because it would have meant pulling the `InjectionContainer` and `InjectionPoint` classes into an SPI module, which seemed too much work and we don't really have a need for that now. 

These methods aren't needed _by us_ at the moment, but since the `BaseRuntime` is extensible, and the `DependencyGraph` is exposed in protected methods, overriders may benefit from them.